dikt/__init__.py

```diff
@@ -39,33 +39,39 @@ def get_chunk_from_key(self, key):
         return self.chunks - 1
 
     def find_key_in_chunk(self, key, chunk):
-        with self.zipf.open(os.path.join(self.name, f"chunk-{chunk:06d}.txt")) as f:
+        with self.zipf.open(f"{self.name}/chunk-{chunk:06d}.txt") as f:
             data = f.read().decode("utf8")
             key = f"K~{key}~"
             start = data.find(key)
             if start == -1:
                 return None
 
             comma = start + len(key)
-            end = data[comma:comma + self.max_len + 1].find("\n")
+            end = data[comma:comma + self.max_len + 1].find("\r")
```

Comment:
what would be the difference with \r instead of \n ?

Reply:
I found that the program would fail to find \n - I kept getting -1 indices. I suggest trying to use your own separator to avoid these issues.

Reply:
What is the OS you're using ? Maybe there's a discrepancy here

Reply:
Windows - but ideally, behaviour should be consistent across operating systems.

Reply:
Of course it should! I'll try with \r to find out if it works with Linux.

Reply:
Do you have a short working example of the \n issue ?